yarn-project/ethereum/src/l1_tx_utils.ts

```diff
@@ -400,6 +400,30 @@ export class L1TxUtils {
       // Check if tx has timed out.
       txTimedOut = isTimedOut();
     }
+
+    // Get the last used gas price if available
+    const lastTx = await this.publicClient.getTransaction({ hash: currentTxHash });
+    const lastGasPrice =
+      lastTx.maxFeePerGas && lastTx.maxPriorityFeePerGas
+        ? {
+            maxFeePerGas: lastTx.maxFeePerGas,
+            maxPriorityFeePerGas: lastTx.maxPriorityFeePerGas,
+            maxFeePerBlobGas: lastTx.maxFeePerBlobGas,
+          }
+        : undefined;
```

Comment:
Let's get the gasPrice for the lastTx from the loop above rather than going to the node again, just in case it fails to fetch it for some reason.